streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamAggregate.java

```diff
@@ -59,7 +61,7 @@ public void enableSendingOldValues() {
     private class KStreamAggregateProcessor extends AbstractProcessor<K, V> {
         private TimestampedKeyValueStore<K, T> store;
         private StreamsMetricsImpl metrics;
-        private Sensor skippedRecordsSensor;
+        private Optional<Sensor> skippedRecordsSensor;
```

Comment:
Maybe keep as `Sensor` and we can set the correct sensor when adding dropped records?  Just a thought I don't have an issue with the current approach.

Reply:
In the task-level metrics refactoring, I will introduce method droppedRecordsSensorOrSkippedRecordsSensor() that chooses which sensor to create depending on the version.